src/iterators/RegionParticleIterator.h

```diff
@@ -10,6 +10,7 @@
 #pragma once
 
 namespace autopas {
+namespace internal {
```

Comment:
i like the internal namespace but we should specify somewhere how to use them / whats in what. Can this be done via doxygen or should we do this via the github wiki (or both)?

Reply:
This is not an general info for the README. I think the best way would be Doxygen so to keep all documentation in one place.

Reply:
I just tested putting a documentation block on top of one namespace line:
```
/**
 * doc for the namespace
 */
namespace autopas { ... }
```

I think what doxygen turns this into is quite useful.

Reply:
ok, somewhere in an extra file? namespaces.h or so?

Reply:
probably a good idea yes.

Reply:
ok, I will add that

Reply:
hmm there are a total of three namespaces that only span one class:
```
namespace arrayMath {
namespace logger {
namespace ThreeDimensionalMapping {
```

at least the logger I would like to convert to a class. What about the others? They are sometimes used via:
```C++
  using ThreeDimensionalMapping::threeToOneD;
```
do we want them to stay namespaces or should we convert them to classes, too?

Reply:
Turning `ThreeDimensionalMapping` into a class `ThreeDimensionalMapper` with static methods sounds the most intuitive to me.

Reply:
ok done (without the renaming though)